lib/datadog/core/configuration/stable_config.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Datadog
+  module Core
+    module Configuration
+      # Import config from config files (fleet automation)
+      module StableConfig
+        module_function
+
+        def extract_configuration
+          if (libdatadog_api_failure = Datadog::Core::LIBDATADOG_API_FAILURE)
+            if Datadog.send(:configuration?)
+              Datadog.logger.debug("Cannot enable stable config: #{libdatadog_api_failure}")
+            else
+              Datadog.send(:logger_without_configuration).debug("Cannot enable stable config: #{libdatadog_api_failure}")
+            end
```

Comment:
```suggestion
            Datadog.safe_logger.debug("Cannot enable stable config: #{libdatadog_api_failure}")
```
Maybe this is a naive suggestion, but can have method so that returns either the fully featured logger, or the fallback without configuration logger? (my method name suggested above is likely bad)

Reply:
Done in [b4e44d5](https://github.com/DataDog/dd-trace-rb/pull/4592/commits/b4e44d5615c5f534053271f1594ba59ff2fe61fd)